src/Components/ExtractLevelsMenu.jsx

```diff
@@ -0,0 +1,184 @@
+
+import React, {useState, useEffect} from 'react'
+import Menu from '@mui/material/Menu'
+import MenuItem from '@mui/material/MenuItem'
+import {makeStyles} from '@mui/styles'
+import {TooltipIconButton} from './Buttons'
+import useStore from '../store/useStore'
+import {getModelCenter} from '../utils/cutPlane'
+import {addHashParams, getHashParams} from '../utils/location'
+import {Vector3} from 'three'
+import {useLocation} from 'react-router-dom'
+import {removePlanes} from '../utils/cutPlane'
+import LevelsIcon from '../assets/2D_Icons/Levels.svg'
+import PlanViewIcon from '../assets/2D_Icons/PlanView.svg'
+import {extractHeight} from '../utils/extractHeight'
+
+/**
+ * BasicMenu used when there are several option behind UI button
+ * show/hide from the right of the screen.
+ *
+ * @param {Array} listOfOptions Title for the drawer
+ * @return {object} ItemPropertiesDrawer react component
+ */
+export default function ExtractLevelsMenu({listOfOptions, icon, title}) {
+  const [anchorEl, setAnchorEl] = useState(null)
+  const classes = useStyles()
+  const open = Boolean(anchorEl)
+
+  const model = useStore((state) => state.modelStore)
+  let [floorplanMenuItems, showExtractMenu] = useState([])
+
+
```

Comment:
remove line

Reply:
Fixed

Reply:
> Looks pretty good! Some comments in there. Can you tell me the status of the impl from ifclib? It would be good to get this and that code merged, esp now that you've seen how you want it here from scratch. Sorry for the duplicate work!

When initially utilizing ifclib, I was hitting multiple runtime errors and was not able narrow down to the problem. Now I can go back and see if I can integrate ifclib with a similar implementation. I do, however, suggest we merge this asap for the ai proposal demo we need to record next week.